fairlearn/metrics/_metric_frame.py

```diff
@@ -461,24 +520,39 @@ def difference(self,
         ----------
         method : str
             How to compute the aggregate. Default is :code:`between_groups`
+        errors: {'raise', 'coerce'}, default 'coerce'
+            if 'raise', then invalid parsing will raise an exception
+            if 'coerce', then invalid parsing will be set as NaN
 
         Returns
         -------
         typing.Any or pandas.Series or pandas.DataFrame
             The exact type follows the table in :attr:`.MetricFrame.overall`.
         """
-        subtrahend = np.nan
+        if errors not in _VALID_ERROR_STRING:
+            raise ValueError(_INVALID_ERRORS_VALUE_ERROR_MESSAGE)
+
         if method == 'between_groups':
-            subtrahend = self.group_min()
+            subtrahend = self.group_min(errors=errors)
         elif method == 'to_overall':
             subtrahend = self.overall
         else:
             raise ValueError("Unrecognised method '{0}' in difference() call".format(method))
 
-        return (self.by_group - subtrahend).abs().max(level=self.control_levels)
+        try:
```

Comment:
We still need another `try` block here? Haven't all possible routes to the exception already been raised, or the values replaced by `NaN`?

Reply:
I found a problem with this piece of code. Cause the code should indeed work with only 
` mf = self.by_group.copy()`
`mf = mf.applymap(lambda x: x if np.isscalar(x) else np.nan)`
`return (mf - subtrahend).abs().max(level=self.control_levels)`

However this doesn't work, since `.applymap()` only works for DataFrames, and Metricframe can sometimes be a Series. So I'm now afraid that the code will not work in the case that Metricframe is a Series and contains a confusion_matrix, so I will first work on new test cases.

Reply:
`MetricFrame.by_group` can be a `Series`? 😮 Is that intended? @riedgar-ms

Reply:
I believe so. In the case where you supply a single metric function, the result is a Series. If you supply a single metric function in a dictionary, it will be a DataFrame with a single column. We had some lengthy debates about this when putting `MetricFrame` together.

Reply:
... but you can have 1-d DataFrames, so what's the benefit of the special case?

Reply:
Because passing a metric function was considered to be different from passing a dictionary containing a single metric function.

Reply:
Believe so, the doc says:
```
@property
def by_group(self) -> Union[pd.Series, pd.DataFrame]:
"""
Returns
        -------
        pandas.Series or pandas.DataFrame
            When a callable is supplied to the constructor, the result is
            a :class:`pandas.Series`, indexed by the combinations of subgroups
            in the sensitive and control features.

            When the metric functions were specified with a dictionary (even
            if the dictionary only has a single entry), then the result is
            a :class:`pandas.DataFrame` with columns named after the metric
            functions, and rows indexed by the combinations of subgroups
            in the sensitive and control features.
```

Reply:
If I recall correctly the reason for this was that indexing a 1-d DataFrame without a sensible column name (which it doesn't have if you only supply a metric function) becomes a bit weird. For example, in order to access the score for some group called 'a' you'd have to do something like `mf.by_group.loc[0,'a']` instead of `mf.by_group.loc['a']`.

Reply:
Yes, it was something like that - the 'without a sensible column name' particularly coming from things constructed with `functools.partial()`. And @adrinjalali soothed my ruffled type-feathers by pointing out that changing from function-dictionary to a single function would in most languages result in an overloaded call, making the change in return types less jarring.